pkg/aws/awsapi.go

```diff
@@ -0,0 +1,54 @@
+package aws
+
+import (
+	"context"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/sqs"
+	"github.com/aws/aws-sdk-go-v2/service/iam"
+	"github.com/aws/aws-sdk-go-v2/service/eks"
+	"github.com/aws/aws-sdk-go-v2/service/pricing"
+	"github.com/aws/aws-sdk-go-v2/service/ssm"
+)
+
+type AWSAPI interface {
```

Comment:
I don't think we want a single interface for all of the services -- let's create separate interfaces per-service.